"""updating models for mapping error

Revision ID: 8e2e0a0da9a0
Revises: 784f4138ef8d
Create Date: 2025-10-06 04:29:24.351535

"""
from alembic import op
import sqlalchemy as sa


# revision identifiers, used by Alembic.
revision = '8e2e0a0da9a0'
down_revision = '784f4138ef8d'
branch_labels = None
depends_on = None


def upgrade():
    # ### commands auto generated by Alembic - please adjust! ###
    with op.batch_alter_table('messages', schema=None) as batch_op:
        batch_op.add_column(sa.Column('user_id', sa.Integer(), nullable=True))

    # Get the id of the first user
    conn = op.get_bind()
    result = conn.execute(sa.text("SELECT id FROM users ORDER BY id LIMIT 1")).fetchone()
    default_user_id = result[0] if result else None

    if default_user_id:
        # Populate existing rows with the default user_id
        op.execute(f'UPDATE messages SET user_id = {default_user_id} WHERE user_id IS NULL')

    with op.batch_alter_table('messages', schema=None) as batch_op:
        batch_op.alter_column('user_id', nullable=False)
        batch_op.create_index(batch_op.f('ix_messages_user_id'), ['user_id'], unique=False)
        batch_op.create_foreign_key(None, 'users', ['user_id'], ['id'])
    # ### end Alembic commands ###


def downgrade():
    # ### commands auto generated by Alembic - please adjust! ###
    with op.batch_alter_table('messages', schema=None) as batch_op:
        batch_op.drop_constraint(None, type_='foreignkey')
        batch_op.drop_index(batch_op.f('ix_messages_user_id'))
        batch_op.drop_column('user_id')

    # ### end Alembic commands ###
